tests/test_cases/test_array/test_array.py

```diff
@@ -247,7 +247,9 @@ def test_gen_loop(dut):
         tlog.info("Iterate access found %s", gens)
 
 
-@cocotb.test()
+@cocotb.test(
+    skip=cocotb.LANGUAGE in ["vhdl"] and cocotb.SIM_NAME.lower().startswith(("riviera"))
+)
```

Comment:
What's the issue with this test? Is the count incorrect, or does it fail in some horrible way?

Reply:
`# # COUT:      6.00ns INFO     cocotb.test                               test_array.py:378  in test_discover_all               Iterating over HierarchyObject(array_module with definition array_module (at /home/az_devops_agent/agent/_work/1/s/tests/designs/array_module/array_module.vhd)) (GPI_MODULE)`

+ SIGSEGV and riviera stuckes

Reply:
Does #1882 fix this?

Reply:
I cannot simply test this (only in CI). I would need to marge this branch here.